### PR TITLE
Request download feature

### PR DESCRIPTION
**Pull request by chrstngyn**

### CODE REVIEW

**Comment by sgfost** [comment]
Currently MemberProfile has `institution = FK(Institution)` and `affiliations = JSONField()`.
A better schema may be:

```python
MemberProfile:
  ...
  # remove institution
  ...
  affiliations = ManyToManyField(Institution, through='Affiliation')
  ...

Affiliation:
  institution = ForeignKey(Institution)
  user = ForeignKey(MemberProfile)
  primary = BooleanField()
  # OR
  rank = IntegerField(min=1, max=10)

Institution:
  name = CharField() # required
  url = URLField() # not required?
  acronym = CharField() # not required
  ror_id = URLField() # not required
```

File: django/core/management/commands/populate_memberprofile_affiliations.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                            # add ror_id if we have a confident enough match
+                            new_affil["ror_id"] = best_match["organization"]["id"]
+                            # add acronym
+                            if best_match["organization"]["acronyms"]:
+                                new_affil["acronym"] = best_match["organization"]["acronyms"][0]
+                            # add link from the response if the user doesn't have one
+                            if not mp.institution.url and best_match["organization"]["links"]:
+                                new_affil["url"] = best_match["organization"]["links"][0]
+
+                    if mp.institution.url:
```
Comment (alee) [inline django/core/management/commands/populate_memberprofile_affiliations.py:48]: consider nesting this in an else: clause of `if best_match:` so we never override any incoming best_match URL

File: django/core/management/commands/populate_memberprofile_affiliations.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            if mp.institution:
+                new_affil = {}
+
+                if mp.institution.name:
+                    new_affil["name"] = mp.institution.name
+                    best_match = self.lookup(mp.institution.name)
+
+                    if best_match:
+                        if (best_match["score"] >= 1.0 and fuzz.partial_ratio(
+                            best_match["organization"]["name"], mp.institution.name) >= 75):
```
Comment (alee) [inline django/core/management/commands/populate_memberprofile_affiliations.py:38]: consider turning the magic constant 75 into an argparse argument with a default value of 75

File: django/core/management/commands/setup_frontend_tests.py [hunk 3]
```diff
@@ -18,9 +18,10 @@ def handle(self, *args, **options):
         user.set_password(settings.TEST_BASIC_AUTH_PASSWORD)
         user.save()
         mp, created = MemberProfile.objects.get_or_create(user=user)
-        mp.institution = Institution.objects.get_or_create(
-            name="ASU", url="https://www.asu.edu"
-        )[0]
+        mp.affiliations = {
```
Comment (alee) [inline django/core/management/commands/setup_frontend_tests.py:21]: should this be `mp.affiliations = [{ "name": "Arizona State University", ... }]`

File: django/home/serializers.py [hunk 4]
```diff
@@ -112,6 +113,32 @@ class MemberProfileSerializer(serializers.ModelSerializer):
     bio = MarkdownField()
     research_interests = MarkdownField()
 
+    def validate_affiliations(self, value):
```
Comment (alee) [inline django/home/serializers.py:116]: this is good, should we also include a Django model validator? https://docs.djangoproject.com/en/4.1/ref/validators/

File: django/home/serializers.py [hunk 5]
```diff
@@ -112,6 +113,32 @@ class MemberProfileSerializer(serializers.ModelSerializer):
...
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": { "type": "string" },
+                    "url": { "type": "string", "format": "uri", "pattern": "^https?://" },
+                    "acronym": { "type": "string" },
+                    "ror_id": { "type": "string", "format": "uri", "pattern": "^https?://ror.org/" }
+                },
+                "required": [ "name" ]
```
Comment (alee) [inline django/home/serializers.py:128]: is `ror_id` also required?

**Review by alee — COMMENTED** [review]
Looks great! Some minor comments included inline

**Comment by alee** [comment]
@all-contributors please add @sgfost for code, design, test

**Pull request merged**